### PR TITLE
feature: add id_attr class attribute for specifing the node_id field

### PR DESCRIPTION
**Pull request by devkral**
# what
add an optional attribute to configure the node_id field named id_field

# Tests
currently untested

### CODE REVIEW

File: strawberry_django_plus/utils/resolvers.py [hunk 1]
```diff
@@ -338,7 +338,12 @@ def resolve_model_nodes(
         qs = origin.get_queryset(qs, info)  # type:ignore
 
     if node_ids is not None:
-        qs = qs.filter(pk__in=[i.node_id if isinstance(i, GlobalID) else i for i in node_ids])
+        id_field = "pk"
+        if hasattr(origin, "id_field"):
+            id_field = origin.id_field
```
Comment (bellini666) [inline strawberry_django_plus/utils/resolvers.py:343]: Oneline solution :)

```suggestion
        id_field = getattr(origin, "id_field", "pk")
```

File: strawberry_django_plus/relay.py [hunk 2]
```diff
@@ -329,10 +329,14 @@ class Node(abc.ABC):
 
     All types that are relay ready should inherit from this interface and
     implement the following methods.
+    Attributes:
```
Comment (bellini666) [inline strawberry_django_plus/relay.py:332]: Add a visual separation here

```suggestion

    Attributes:
```

File: strawberry_django_plus/utils/resolvers.py [hunk 3]
```diff
@@ -400,7 +405,11 @@ def resolve_model_node(source, node_id, *, info: Optional[Info] = None, required
     if isinstance(node_id, GlobalID):
         node_id = node_id.node_id
 
-    qs = source._default_manager.filter(pk=node_id)
+    id_field = "pk"
+    if hasattr(origin, "id_field"):
+        id_field = origin.id_field
```
Comment (bellini666) [inline strawberry_django_plus/utils/resolvers.py:410]: Ditto

```suggestion
    id_field = getattr(origin, "id_field", "pk")
```

File: strawberry_django_plus/relay.py [hunk 4]
```diff
@@ -329,10 +329,14 @@ class Node(abc.ABC):
...
+        id_field:
```
Comment (bellini666) [inline strawberry_django_plus/relay.py:333]: What do you think of calling it `id_attr`? To be explicit that it is the attribute to retrieve the id and not a possibly strawberry field.

**Review by bellini666 — CHANGES_REQUESTED** [review]
Thanks for the contribution :)

Just a small suggestions and this should be good to merge

**Comment by devkral** [comment]
I have some problems writing tests:

how can I create a node where I can set the attribute? The factories automate everything

File: strawberry_django_plus/relay.py (hunk 4 above)
Comment (devkral) [reply strawberry_django_plus/relay.py:334]: sounds good

**Comment by bellini666** [comment]
> how can I create a node where I can set the attribute? The factories automate everything

Sorry, I don't know if I understood what you want to do

**Comment by devkral** [comment]
I want to add a test for the id_attr behaviour.
For this I would need something like:

```python
@gql.django.type(Foo)
class FooNode(relay.Node):
     id_attr = 'name'
```

**Comment by bellini666** [comment]
Oh I see. For simplicity I'm using the demo project for tests, but you could create your type/schema directly inside the tests file itself.

**Comment by devkral** [comment]
would you mind if I change the demo user to have its username as node id?

This would serve as a fine example as well as being a good test

**Comment by bellini666** [comment]
> would you mind if I change the demo user to have its username as node id?
> 
> This would serve as a fine example as well as being a good test

Not at all, go ahead :)

File: strawberry_django_plus/utils/resolvers.py [hunk 5]
```diff
@@ -413,25 +418,29 @@ def resolve_model_node(source, node_id, *, info: Optional[Info] = None, required
...
     Returns:
         The resolved object id
 
     """
+    id_attr = getattr(source, "id_attr", "pk")
     assert isinstance(root, Model)
-    pk = root.__class__._meta.pk
-    assert pk
+    if id_attr == "pk":
+        id_attr = root.__class__._meta.pk.attname
```
Comment (bellini666) [inline strawberry_django_plus/utils/resolvers.py:437]: You will probably need to do this to fix pyright issue (the typing says `pk` can be `None`)

```suggestion
        pk = root.__class__._meta.pk
        assert pk
        id_attr = pk.attname
```

**Review by bellini666 — CHANGES_REQUESTED** [review]
Looks good! Just a small typing issue and should be fine to merge

File: strawberry_django_plus/utils/resolvers.py (hunk 5 above)
Comment (devkral) [reply strawberry_django_plus/utils/resolvers.py:437]: thanks, will include this

**Review by bellini666 — APPROVED** [review]
LGTM 👍🏼 
Thank you!

**Pull request merged**